integrationTests/vm/arwen/utils.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract"
+	"github.com/stretchr/testify/assert"
+)
+
+type testContext struct {
+	T *testing.T
+
+	Round uint64
+
+	Owner testParticipant
+	Alice testParticipant
+	Bob   testParticipant
+	Carol testParticipant
+
+	ScAddress    []byte
+	Accounts     *state.AccountsDB
+	TxProcessor  process.TransactionProcessor
+	QueryService external.SCQueryService
+}
+
+type testParticipant struct {
+	Nonce   uint64
+	Address []byte
+	Balance *big.Int
+}
+
+func (participant *testParticipant) AddressHex() string {
+	return hex.EncodeToString(participant.Address)
+}
+
+func setupTestContext(t *testing.T) testContext {
+	context := testContext{}
+	context.T = t
+	context.Round = 500
+
+	context.initAccounts()
+
+	gasSchedule, err := core.LoadGasScheduleConfig("./gasSchedule.toml")
+	assert.Nil(t, err)
+
+	vmContainer, blockChainHook := vm.CreateVMAndBlockchainHook(context.Accounts, gasSchedule)
+	context.TxProcessor = vm.CreateTxProcessorWithOneSCExecutorWithVMs(context.Accounts, vmContainer, blockChainHook)
+	context.ScAddress, _ = blockChainHook.NewAddress(context.Owner.Address, context.Owner.Nonce, factory.ArwenVirtualMachine)
+	context.QueryService, _ = smartContract.NewSCQueryService(vmContainer, math.MaxInt32)
+	return context
+}
+
+func (context *testContext) initAccounts() {
+	context.Owner = testParticipant{}
+	context.Owner.Address = []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o', 'o'}
+	context.Owner.Nonce = uint64(1)
+	context.Owner.Balance = big.NewInt(math.MaxInt64)
+
+	context.Alice = testParticipant{}
+	context.Alice.Address = []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a', 'a'}
+	context.Alice.Nonce = uint64(1)
+	context.Alice.Balance = big.NewInt(math.MaxInt64)
+
+	context.Bob = testParticipant{}
+	context.Bob.Address = []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b', 'b'}
+	context.Bob.Nonce = uint64(1)
+	context.Bob.Balance = big.NewInt(math.MaxInt64)
+
+	context.Carol = testParticipant{}
+	context.Carol.Address = []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c', 'c'}
+	context.Carol.Nonce = uint64(1)
+	context.Carol.Balance = big.NewInt(math.MaxInt64)
+
+	context.Accounts = vm.CreateInMemoryShardAccountsDB()
+	context.createAccount(&context.Owner)
+	context.createAccount(&context.Alice)
+	context.createAccount(&context.Bob)
+	context.createAccount(&context.Carol)
+}
+
+func (context *testContext) createAccount(participant *testParticipant) {
+	_ = vm.CreateAccount(context.Accounts, participant.Address, participant.Nonce, participant.Balance)
```

Comment:
maybe you can put an assert here to check if an error appears

Reply:
Added a check.